portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -26,23 +26,52 @@
 import com.google.gson.Gson;
 import com.google.gson.internal.GsonBuildConfig;
 
-import com.google.sps.data.Comments;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+
+import com.google.sps.data.Comment;
 
 /** Servlet that returns some example content. */
 @WebServlet("/data")
 public final class DataServlet extends HttpServlet {
-    private Comments comments = new Comments();
-
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        response.setContentType("application/json");
-        String json = new Gson().toJson(comments);
-        response.getWriter().println(json);
+        Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        PreparedQuery results = datastore.prepare(query);
+
+        List<Comment> comments = new ArrayList<>();
+        for (Entity entity : results.asIterable()) {
+            String comment = (String) entity.getProperty("comment");
+            long timestamp = (long) entity.getProperty("timestamp");
+
+            Comment nextComment = new Comment(comment, timestamp);
```

Comment:
Prefer to use clearer names such as:

String content = (String) entity.getProperty("comment");
long timestamp = ...

Comment comment = new Comment(content, timestmap);
comments.add(comment);

Reply:
Also done